web/src/settings.ts

```diff
@@ -155,6 +182,12 @@ export function build_page(): void {
     });
 
     $(".settings-box").html(rendered_settings_tab);
+    // Set the default role of the user.
+    const current_user_role = people.get_by_user_id(current_user.user_id).role;
+    $("#user-self-role-select").val(current_user_role);
+    if (!current_user.is_owner) {
+        remove_owner_from_role_select();
```

Comment:
This code should be in `settings_account.set_up` where we initialize other elements on the page.